compiler/luci/service/src/Nodes/CircleReshape.test.cpp

```diff
@@ -197,3 +197,40 @@ TEST(ShapeRuleTest, reshape_by_newShape)
   ASSERT_EQ(2, output_shape.dim(0).value());
   ASSERT_EQ(12, output_shape.dim(1).value());
 }
+
+TEST(ShapeRuleTest, reshape_by_newShape_dynamic)
+{
+  auto g = loco::make_graph();
+  auto node_reshape = g->nodes()->create<luci::CircleReshape>();
+  auto tensor_input = g->nodes()->create<luci::CircleInput>();
+  auto target_shape = g->nodes()->create<luci::CircleInput>();
+  ;
+
+  tensor_input->dtype(loco::DataType::S32);
+  tensor_input->shape({2, 3, 4});
+  tensor_input->shape_status(luci::ShapeStatus::VALID);
+
+  target_shape->dtype(loco::DataType::S32);
+  target_shape->rank(1);
+  target_shape->shape_status(luci::ShapeStatus::VALID);
+
+  node_reshape->tensor(tensor_input);
+  node_reshape->shape(target_shape);
+
+  // reshape to {dynamic, 4, dynamic}
+  node_reshape->newShape()->rank(3);
+  node_reshape->newShape()->dim(0) = -1;
+  node_reshape->newShape()->dim(1) = 4;
+  node_reshape->newShape()->dim(2) = -1;
+
+  loco::TensorShape output_shape;
+  luci::sinf::Rule shape_inf_rule;
+
+  ASSERT_TRUE(shape_inf_rule.infer(node_reshape, output_shape));
+
+  ASSERT_EQ(3, output_shape.rank());
+  ASSERT_FALSE(output_shape.dim(0).known());
+  ASSERT_TRUE(output_shape.dim(1).known());
+  ASSERT_EQ(4, output_shape.dim(1).value());
+  ASSERT_FALSE(output_shape.dim(2).known());
+}
```

Comment:
IIUC, you are to reshape `(2, 3, 4)` to `(-1, 4, -1)`
I understand you are just preparing a test but 
I really don't understand how `(2, 3, 4)` can be reshaped to `(-1, 4, -1)`.
This values can make others reading this code give confusion.
If you think I don't know about this, please explain.

Reply:
it's probably can be reshaping to `(6, 4, 1)` but agree that such test data can be confusing. Changed to `2` (still with assumption that -1 means dyn dimension)